### ISSUE TITLE
use_g_object_notify_by_pspec shouldn't warn about some properties

### ISSUE THREAD

**Issue by benjamin-otte** [bilelmoussaoui/gobject-lint#103]
Properties that are defined in an external file (in the GTK case this happens in the GDK backends which are meant to `g_object_notify()` the parent class's properties) and overridden properties (because their pspec doesn't even get exposed usually) should probably not cause the `use_g_object_notify_by_pspec` rule to trigger.

The rule doesn't fix those cases either.

There's about 50 - 100 properties in GTK that fall into those 2 categories and prevent us from using this rule.

**Comment by bilelmoussaoui** [comment]
Right, so you would expect that only properties defined in the same file to be flagged? what about cases where you would have a `my_property_notify` or that never happens?

**Comment by benjamin-otte** [comment]
> Right, so you would expect that only properties defined in the same file to be flagged?

In modern code in GTK, we have a private setter function for all properties, so notifies only happen in the same file.

But there's a bunch of rather old code - mainly around `gdkdevice.c` - where `gdkwaylanddevice.c` is supposed to manage and notify the property. It works a bit like an interface conceptually, but it's a base class in code.

> what about cases where you would have a `my_property_notify` or that never happens?

What do you mean with `my_property_notify()`?

**Comment by bilelmoussaoui** [comment]
In Rust where the properties boilerplate is generated by a macro, it also generates a helper function called property_name_notify so you don't mispell the property name elsewhere

**Comment by benjamin-otte** [comment]
fwiw, you can just run `use_gobject_class_install_properties --fix` and then `use_g_object_notify_by_pspec --fix` on the GTK source code.

I'm getting about 200 remaining calls to g_object_notify() and they seem to fall into 4 big categories:

1. The `gdkdevice.c` case with external files

2. The `override_property()` case

3. GTK's `ADD_SET_PROP()` macro and other weird stuff in old GTK which I'm working on getting rid of atm.

4. Some cases that look obvious yet the tool fails to fix things, I have no idea why.

**Comment by benjamin-otte** [comment]
> In Rust where the properties boilerplate is generated by a macro, it also generates a helper function called property_name_notify so you don't mispell the property name elsewhere

GTK does not do that at all.

Old code just used `g_object_notify()` and in new code we have the enum and the compiler complains if we get the enum wrong.

**Comment by bilelmoussaoui** [comment]
I guess you could add gdkdevice.c and other files to the ignore list of that rule , except for the override case?

For cases where the tool fails, please let me know which files are failing so i can look at them and figure out what is going wrong. I test the rule on mutter/gnome-shell/gdm only

**Comment by benjamin-otte** [comment]
An example it fails to fix is [this one](https://gitlab.gnome.org/GNOME/gtk/-/blob/c9f298305faf3002dfabb3e426cd1c8846429ca3/gtk/print/backends/gtkprintercups.c#L281) - I have no idea why.

It also seems to fail with files that have >1 object like [this one](https://gitlab.gnome.org/GNOME/gtk/-/blob/c9f298305faf3002dfabb3e426cd1c8846429ca3/testsuite/gtk/filterlistmodel.c) - even though each property only appears in one of the 2 enums, so it could technically handle it.

**Comment by bilelmoussaoui** [comment]
Both are now detected and fixable properly.

**Comment by bilelmoussaoui** [comment]
We now skip override propeties as well until there is a solution for that in the other issue you reported.

**Comment by benjamin-otte** [comment]
Ran git main on GTK, and for some reason it seems to skip the whole `gtk/deprecated` folder - [gtkcellrenderer.c](https://gitlab.gnome.org/GNOME/gtk/-/blob/wip/otte/goblint/gtk/deprecated/gtkcellrenderer.c) is a file I manually made sure to have all properties correct put into a `props` array, yet it `--fix` doesn't fix the notify calls. It does mark them as violations though.

**Comment by bilelmoussaoui** [comment]
Let me try that.

**Comment by bilelmoussaoui** [comment]
They are now all fixable except few cases: 
```
Found 142 error(s)
/home/user/Projects/gtk/gdk/android/gdkandroidtoplevel.c:422:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_TITLE]) instead of g_object_notify(obj, "title") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/gdksurface.c:1049:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_STATE]) instead of g_object_notify(obj, "state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/gdksurface.c:2753:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_STATE]) instead of g_object_notify(obj, "state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/gdksurface.c:2756:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_STICKY]) instead of g_object_notify(obj, "sticky") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1399:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_DIRECTION]) instead of g_object_notify(obj, "direction") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1401:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_HAS_BIDI_LAYOUTS]) instead of g_object_notify(obj, "has-bidi-layouts") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1403:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_CAPS_LOCK_STATE]) instead of g_object_notify(obj, "caps-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1405:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_NUM_LOCK_STATE]) instead of g_object_notify(obj, "num-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1407:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_SCROLL_LOCK_STATE]) instead of g_object_notify(obj, "scroll-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1409:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_MODIFIER_STATE]) instead of g_object_notify(obj, "modifier-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1411:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_ACTIVE_LAYOUT_INDEX]) instead of g_object_notify(obj, "active-layout-index") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1414:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_LAYOUT_NAMES]) instead of g_object_notify(obj, "layout-names") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1580:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_ACTIVE_LAYOUT_INDEX]) instead of g_object_notify(obj, "active-layout-index") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1585:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_DIRECTION]) instead of g_object_notify(obj, "direction") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1588:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_HAS_BIDI_LAYOUTS]) instead of g_object_notify(obj, "has-bidi-layouts") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1590:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_CAPS_LOCK_STATE]) instead of g_object_notify(obj, "caps-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1592:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_NUM_LOCK_STATE]) instead of g_object_notify(obj, "num-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1594:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_SCROLL_LOCK_STATE]) instead of g_object_notify(obj, "scroll-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1596:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_MODIFIER_STATE]) instead of g_object_notify(obj, "modifier-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdkseat-wayland.c:1599:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_LAYOUT_NAMES]) instead of g_object_notify(obj, "layout-names") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdksurface-wayland.c:241:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_WIDTH]) instead of g_object_notify(obj, "width") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdksurface-wayland.c:243:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_HEIGHT]) instead of g_object_notify(obj, "height") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdksurface-wayland.c:245:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_SCALE]) instead of g_object_notify(obj, "scale") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdksurface-wayland.c:247:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_SCALE_FACTOR]) instead of g_object_notify(obj, "scale-factor") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdktoplevel-wayland.c:867:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_CAPABILITIES]) instead of g_object_notify(obj, "capabilities") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdktoplevel-wayland.c:994:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_CAPABILITIES]) instead of g_object_notify(obj, "capabilities") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdktoplevel-wayland.c:1456:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_DECORATED]) instead of g_object_notify(obj, "decorated") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdktoplevel-wayland.c:1876:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUTS_INHIBITED]) instead of g_object_notify(obj, "shortcuts-inhibited") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdktoplevel-wayland.c:1887:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUTS_INHIBITED]) instead of g_object_notify(obj, "shortcuts-inhibited") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/wayland/gdktoplevel-wayland.c:1984:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUTS_INHIBITED]) instead of g_object_notify(obj, "shortcuts-inhibited") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/win32/gdkkeys-win32.c:660:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_ACTIVE_LAYOUT_INDEX]) instead of g_object_notify(obj, "active-layout-index") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/win32/gdkkeys-win32.c:661:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_LAYOUT_NAMES]) instead of g_object_notify(obj, "layout-names") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/win32/gdkkeys-win32.c:693:15: error: Use g_object_notify_by_pspec(obj, properties[PROP_ACTIVE_LAYOUT_INDEX]) instead of g_object_notify(obj, "active-layout-index") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/win32/gdksurface-win32.c:3404:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUTS_INHIBITED]) instead of g_object_notify(obj, "shortcuts-inhibited") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/win32/gdksurface-win32.c:3421:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUTS_INHIBITED]) instead of g_object_notify(obj, "shortcuts-inhibited") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdkkeys-x11.c:626:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_DIRECTION]) instead of g_object_notify(obj, "direction") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdkkeys-x11.c:663:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_CAPS_LOCK_STATE]) instead of g_object_notify(obj, "caps-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdkkeys-x11.c:665:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_NUM_LOCK_STATE]) instead of g_object_notify(obj, "num-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdkkeys-x11.c:667:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_SCROLL_LOCK_STATE]) instead of g_object_notify(obj, "scroll-lock-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdkkeys-x11.c:669:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_MODIFIER_STATE]) instead of g_object_notify(obj, "modifier-state") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdkkeys-x11.c:709:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_ACTIVE_LAYOUT_INDEX]) instead of g_object_notify(obj, "active-layout-index") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdkkeys-x11.c:755:9: error: Use g_object_notify_by_pspec(obj, properties[PROP_LAYOUT_NAMES]) instead of g_object_notify(obj, "layout-names") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdksurface-x11.c:2042:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SCALE_FACTOR]) instead of g_object_notify(obj, "scale-factor") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdksurface-x11.c:2043:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SCALE]) instead of g_object_notify(obj, "scale") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdksurface-x11.c:3564:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FULLSCREEN_MODE]) instead of g_object_notify(obj, "fullscreen-mode") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdksurface-x11.c:5447:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUTS_INHIBITED]) instead of g_object_notify(obj, "shortcuts-inhibited") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdksurface-x11.c:5464:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUTS_INHIBITED]) instead of g_object_notify(obj, "shortcuts-inhibited") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gdk/x11/gdksurface-x11.c:5495:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUTS_INHIBITED]) instead of g_object_notify(obj, "shortcuts-inhibited") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkcellrenderer.c:571:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_CELL_BACKGROUND_SET]) instead of g_object_notify(obj, "cell-background-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkcellrenderer.c:591:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_CELL_BACKGROUND_SET]) instead of g_object_notify(obj, "cell-background-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkcellrenderer.c:601:4: error: Use g_object_notify_by_pspec(obj, properties[PROP_CELL_BACKGROUND_SET]) instead of g_object_notify(obj, "cell-background-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkcellrendereraccel.c:403:11: error: Use g_object_notify_by_pspec(obj, gtk_cell_renderer_accel_props[PROP_ACCEL_MODE]) instead of g_object_notify(obj, "accel-mode") for better performance (ambiguous: multiple classes define this property) [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkfontbutton.c:331:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHOW_PREVIEW_ENTRY]) instead of g_object_notify(obj, "show-preview-entry") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkfontbutton.c:416:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT]) instead of g_object_notify(obj, "font") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkfontbutton.c:417:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT_DESC]) instead of g_object_notify(obj, "font-desc") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkfontbutton.c:1091:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT]) instead of g_object_notify(obj, "font") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkfontbutton.c:1092:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT_DESC]) instead of g_object_notify(obj, "font-desc") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkfontbutton.c:1093:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT_FEATURES]) instead of g_object_notify(obj, "font-features") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkfontbutton.c:1203:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_LEVEL]) instead of g_object_notify(obj, "level") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/deprecated/gtkfontbutton.c:1215:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_LANGUAGE]) instead of g_object_notify(obj, "language") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkactionhelper.c:479:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_ACTION_NAME]) instead of g_object_notify(obj, "action-name") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkactionhelper.c:555:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_ACTION_TARGET]) instead of g_object_notify(obj, "action-target") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfilechoosernative.c:457:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_FILTER]) instead of g_object_notify(obj, "filter") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfilechoosernativequartz.c:111:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FILTER]) instead of g_object_notify(obj, "filter") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfilechooserwidget.c:2602:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SELECT_MULTIPLE]) instead of g_object_notify(obj, "select-multiple") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfilechooserwidget.c:4855:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FILTERS]) instead of g_object_notify(obj, "filters") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfilechooserwidget.c:4889:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FILTERS]) instead of g_object_notify(obj, "filters") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfilechooserwidget.c:4909:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUT_FOLDERS]) instead of g_object_notify(obj, "shortcut-folders") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfilechooserwidget.c:4923:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHORTCUT_FOLDERS]) instead of g_object_notify(obj, "shortcut-folders") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfilechooserwidget.c:6083:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_FILTER]) instead of g_object_notify(obj, "filter") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:667:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT]) instead of g_object_notify(obj, "font") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:668:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT_DESC]) instead of g_object_notify(obj, "font-desc") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:2821:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT_FEATURES]) instead of g_object_notify(obj, "font-features") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:2879:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT]) instead of g_object_notify(obj, "font") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:2880:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT_DESC]) instead of g_object_notify(obj, "font-desc") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:2924:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_PREVIEW_TEXT]) instead of g_object_notify(obj, "preview-text") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:2943:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHOW_PREVIEW_ENTRY]) instead of g_object_notify(obj, "show-preview-entry") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:3050:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_LEVEL]) instead of g_object_notify(obj, "level") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkfontchooserwidget.c:3091:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_LANGUAGE]) instead of g_object_notify(obj, "language") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkpasswordentrybuffer.c:149:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_TEXT]) instead of g_object_notify(obj, "text") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkpasswordentrybuffer.c:150:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_LENGTH]) instead of g_object_notify(obj, "length") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkpopover.c:2197:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_HAS_DEFAULT]) instead of g_object_notify(obj, "has-default") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkpopover.c:2206:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_HAS_DEFAULT]) instead of g_object_notify(obj, "has-default") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:4014:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_TEXT]) instead of g_object_notify(obj, "text") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:4015:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_COMPLETE_TEXT]) instead of g_object_notify(obj, "complete-text") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:4618:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_COMPLETE_TEXT]) instead of g_object_notify(obj, "complete-text") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:4716:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_CURSOR_POSITION]) instead of g_object_notify(obj, "cursor-position") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:4725:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_SELECTION_BOUND]) instead of g_object_notify(obj, "selection-bound") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:5812:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_TEXT]) instead of g_object_notify(obj, "text") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:5867:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_EDITABLE]) instead of g_object_notify(obj, "editable") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:6247:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_WIDTH_CHARS]) instead of g_object_notify(obj, "width-chars") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:6261:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_MAX_WIDTH_CHARS]) instead of g_object_notify(obj, "max-width-chars") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:6306:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_XALIGN]) instead of g_object_notify(obj, "xalign") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktext.c:7714:3: error: Use g_object_notify_by_pspec(obj, properties[PROP_ENABLE_UNDO]) instead of g_object_notify(obj, "enable-undo") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1091:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_UNDERLINE_RGBA_SET]) instead of g_object_notify(obj, "underline-rgba-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1099:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_UNDERLINE_RGBA_SET]) instead of g_object_notify(obj, "underline-rgba-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1121:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_OVERLINE_RGBA_SET]) instead of g_object_notify(obj, "overline-rgba-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1129:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_OVERLINE_RGBA_SET]) instead of g_object_notify(obj, "overline-rgba-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1151:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_STRIKETHROUGH_RGBA_SET]) instead of g_object_notify(obj, "strikethrough-rgba-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1159:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_STRIKETHROUGH_RGBA_SET]) instead of g_object_notify(obj, "strikethrough-rgba-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1178:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_BACKGROUND_SET]) instead of g_object_notify(obj, "background-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1188:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_BACKGROUND_SET]) instead of g_object_notify(obj, "background-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1207:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_FOREGROUND_SET]) instead of g_object_notify(obj, "foreground-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1217:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_FOREGROUND_SET]) instead of g_object_notify(obj, "foreground-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1236:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_PARAGRAPH_BACKGROUND_SET]) instead of g_object_notify(obj, "paragraph-background-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1246:11: error: Use g_object_notify_by_pspec(obj, properties[PROP_PARAGRAPH_BACKGROUND_SET]) instead of g_object_notify(obj, "paragraph-background-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1318:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_FAMILY_SET]) instead of g_object_notify(obj, "family-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1320:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_STYLE_SET]) instead of g_object_notify(obj, "style-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1322:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_VARIANT_SET]) instead of g_object_notify(obj, "variant-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1324:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_WEIGHT_SET]) instead of g_object_notify(obj, "weight-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1326:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_STRETCH_SET]) instead of g_object_notify(obj, "stretch-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1328:5: error: Use g_object_notify_by_pspec(obj, properties[PROP_SIZE_SET]) instead of g_object_notify(obj, "size-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1562:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_SCALE_SET]) instead of g_object_notify(obj, "scale-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1569:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_PIXELS_ABOVE_LINES_SET]) instead of g_object_notify(obj, "pixels-above-lines-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1576:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_PIXELS_BELOW_LINES_SET]) instead of g_object_notify(obj, "pixels-below-lines-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1583:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_PIXELS_INSIDE_WRAP_SET]) instead of g_object_notify(obj, "pixels-inside-wrap-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1590:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_LINE_HEIGHT_SET]) instead of g_object_notify(obj, "line-height-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1597:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_EDITABLE_SET]) instead of g_object_notify(obj, "editable-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1603:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_WRAP_MODE_SET]) instead of g_object_notify(obj, "wrap-mode-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1610:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_JUSTIFICATION_SET]) instead of g_object_notify(obj, "justification-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1621:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_LEFT_MARGIN_SET]) instead of g_object_notify(obj, "left-margin-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1628:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_INDENT_SET]) instead of g_object_notify(obj, "indent-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1635:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_STRIKETHROUGH_SET]) instead of g_object_notify(obj, "strikethrough-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1648:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_RIGHT_MARGIN_SET]) instead of g_object_notify(obj, "right-margin-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1655:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_UNDERLINE_SET]) instead of g_object_notify(obj, "underline-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1668:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_OVERLINE_SET]) instead of g_object_notify(obj, "overline-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1681:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_RISE_SET]) instead of g_object_notify(obj, "rise-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1688:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_BACKGROUND_FULL_HEIGHT_SET]) instead of g_object_notify(obj, "background-full-height-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1694:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_LANGUAGE_SET]) instead of g_object_notify(obj, "language-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1707:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_TABS_SET]) instead of g_object_notify(obj, "tabs-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1715:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_INVISIBLE_SET]) instead of g_object_notify(obj, "invisible-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1743:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_FALLBACK_SET]) instead of g_object_notify(obj, "fallback-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1749:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_LETTER_SPACING_SET]) instead of g_object_notify(obj, "letter-spacing-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1755:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_FONT_FEATURES_SET]) instead of g_object_notify(obj, "font-features-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1761:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_ALLOW_BREAKS_SET]) instead of g_object_notify(obj, "allow-breaks-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1767:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_SHOW_SPACES_SET]) instead of g_object_notify(obj, "show-spaces-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1773:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_INSERT_HYPHENS_SET]) instead of g_object_notify(obj, "insert-hyphens-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1779:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_TEXT_TRANSFORM_SET]) instead of g_object_notify(obj, "text-transform-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1785:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_WORD_SET]) instead of g_object_notify(obj, "word-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtktexttag.c:1791:7: error: Use g_object_notify_by_pspec(obj, properties[PROP_SENTENCE_SET]) instead of g_object_notify(obj, "sentence-set") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkwindow.c:2520:2: error: Use g_object_notify_by_pspec(obj, properties[PROP_HAS_DEFAULT]) instead of g_object_notify(obj, "has-default") for better performance [use_g_object_notify_by_pspec]
/home/user/Projects/gtk/gtk/gtkwindow.c:2524:4: error: Use g_object_notify_by_pspec(obj, properties[PROP_HAS_DEFAULT]) instead of g_object_notify(obj, "has-default") for better performance [use_g_object_notify_by_pspec]
```

Except for cases like `gdk_toplevel_install_properties` and  `ADD_SET_PROP` where we can't follow properties installation properly. 

I guess we should be good here as well
